### PR TITLE
feat: oauth2-proxy helpers (DT-3285)

### PR DESCRIPTION
**Pull request by jaredallard**
**What this PR does**: This PR adds oauth2-proxy helper functions for securing a service behind an oauth2-proxy instance.

**JIRA ID**: [DT-3285]

[DT-3285]: https://outreach-io.atlassian.net/browse/DT-3285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kubernetes/oauth2-proxy.jsonnet [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    backend: {
+      serviceName: serviceName,
+      servicePort: 'http',
+    },
+  },
+
+  // ALBIngressAnnotations returns a set of annotations that are helpful when
+  // using the oauth2-proxy with an ALB Ingress Controller.
+  ALBIngressAnnotations():: {
+    'alb.ingress.kubernetes.io/load-balancer-attributes'+: ',routing.http.preserve_host_header.enabled=true',
```
Comment (kaldorn) [inline kubernetes/oauth2-proxy.jsonnet:114]: I'm curious about what this usage will look like.

**Review by kaldorn — APPROVED** [review]
Looks good. Having trouble fully grokking how that annotation will merge in.

File: kubernetes/oauth2-proxy.jsonnet (hunk 1 above)
Comment (jaredallard) [reply kubernetes/oauth2-proxy.jsonnet:114]: The usage would be `existingAnnotations + ALBIngressAnnotations`, right?

Comment (kaldorn) [reply kubernetes/oauth2-proxy.jsonnet:114]: That looks like it works in the jsonnet repl.

**Pull request merged**